### PR TITLE
#66: Make SSL certificate validation configurable

### LINKED ISSUE #66: Make TLS certificate validation configurable
EPTF generates file `odbc.ini` containing the configuration options for the ODBC driver. 

Newer versions of Exasol database validate the TLS certificates and may cause an error when using a self-signed certificate. 

Such error messages can be suppressed by adding option `;FINGERPRINT=NoCertCheck` to the DSN in file `odbc.ini`.
This option cannot be appended to CLI option `--server`.

### Acceptance Criteria 

The current ticket requests to add a new CLI option to EPTF controlling whether this option should be added.

Additionally, in [client_setup.py#L61](https://github.com/exasol/exasol-python-test-framework/blob/master/exasol_python_test_framework/exatest/clients/client_setup.py#L61) the suffix `.logdir` should be removed.

### LINKED ISSUE #70: Fix vulnerabilities
* [ ] Use latest drivers and ExaPlus from the URLS behind [downloads.exasol.com](https://downloads.exasol.com/clients-and-drivers/odbc)
  * Old: https://exasol-script-languages-dependencies.s3.eu-central-1.amazonaws.com/EXASOL_ODBC-7.0.11.tar.gz
  * New: https://x-up.s3.amazonaws.com/7.x/24.1.1/Exasol_ODBC-24.1.1-Linux_x86_64.tar.gz
* [ ] Fix vulnerabilities 
  * CVE-2024-35195 in dependency `requests` in versions below `2.32.0` caused by requests `Session` object not verifying requests after making first request with `verify=False`
  * CVE-2024-37891 in transitive dependency via `boto3` to `urllib3` in versions below `2.2.2` caused by proxy-authorization request header not to be stripped during cross-origin redirects as no update of notebook-connector is available, yet.
  * GHSA-w235-7p84-xx57 in transitive dependency via `luigi` to `tornado` in versions below `6.4.1` enabling CRLF injection in `CurlAsyncHTTPClient` headers.
  * GHSA-753j-mpmx-qq6g in transitive dependency via `luigi` to `tornado` in versions below `6.4.1` due to inconsistent interpretation of HTTP Requests ('HTTP Request/Response Smuggling')
* [x] Ignore vulnerabilities 
  * GHSA-753j-mpmx-qq6g in dependency `configobj` in versions <= `5.0.8` being ReDoS exploitable by developers using values in a server-side configuration file as SLCT is used only client side and pathed version are not available, yet.

### PR DESCRIPTION
**Pull request by ckunki**
Closes #66 
Closes #70

### All Submissions:

* [x] Is the title of the Pull Request correct?
* [x] Is the title of the corresponding issue correct?
* [x] Have you updated the changelog?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../../pulls) for the same update/change?  
* [x] Are you mentioning the issue which this PullRequest fixes ("Fixes...")

### CODE REVIEW

File: exasol_python_test_framework/exatest/__init__.py [hunk 1]
```diff
@@ -247,6 +247,7 @@ def _parse_opts(self):
                 logdir='.',
                 loglevel='warning',
                 odbc_log='off',
+                ssl_cert='unverified',
```
Comment (kaklakariada) [inline exasol_python_test_framework/exatest/__init__.py:250]:
```suggestion
                tls_cert='unverified',
```

File: exasol_python_test_framework/exatest/__init__.py [hunk 2]
```diff
@@ -282,8 +283,15 @@ def prepare_hook(self):
...
-        self._client_setup.prepare_odbc_init(self.opts.logdir, self.opts.server, self.opts.driver,
-                                             self.opts.user, self.opts.password, self.opts.odbc_log)
+        self._client_setup.prepare_odbc_init(
+            self.opts.logdir,
+            self.opts.server,
+            self.opts.driver,
+            self.opts.user,
+            self.opts.password,
+            self.opts.odbc_log,
+            self.opts.ssl_cert,
```
Comment (kaklakariada) [inline exasol_python_test_framework/exatest/__init__.py:293]:
```suggestion
            self.opts.tls_cert,
```

File: doc/changes/changes_0.6.0.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+Code name: Support for newer Exasol ODBC driver
+
+## Summary
+
+This release supports adding a fingerprint to file `odbc.ini`.
+
+## Features
+
+* #66: Made SSL certificate validation in file `odbc.ini` configurable
```
Comment (kaklakariada) [inline doc/changes/changes_0.6.0.md:11]:
```suggestion
* #66: Made TLS certificate validation in file `odbc.ini` configurable
```

File: exasol_python_test_framework/exatest/clients/client_setup.py [hunk 4]
```diff
@@ -25,47 +42,84 @@ class ClientSetup(object):
...
+            help="path to ODBC driver",
+            required=True,
+            type=validate_driver_path)
+        parser.add_argument(
+            "--odbc-log", choices=LOG_LEVELS,
+            help='activate ODBC driver log (default: %(default)s)',
+        )
+        parser.add_argument(
+            "--ssl-cert", choices=SSL_CERT_OPTIONS,
+            help='SSL certificate verification (default: %(default)s)',
```
Comment (kaklakariada) [inline exasol_python_test_framework/exatest/clients/client_setup.py:66]:
```suggestion
            "--tls-cert", choices=TLS_CERT_OPTIONS,
            help='TLS certificate verification (default: %(default)s)',
```

File: exasol_python_test_framework/exatest/clients/client_setup.py [hunk 5]
```diff
@@ -25,47 +42,84 @@ class ClientSetup(object):
...
+                    [{self.dsn}]
+                    Driver = {driver}
+                    EXAHOST = {server}
+                    EXAUID = {user}
+                    EXAPWD = {password}
+                    CONNECTIONLCCTYPE = en_US.UTF-8
+                    CONNECTIONLCNUMERIC = en_US.UTF-8
+                    """
+            ))
+            print(f'{ssl_cert}')
```
Comment (kaklakariada) [inline exasol_python_test_framework/exatest/clients/client_setup.py:94]: Either use logging or remove this
```suggestion
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: exasol_python_test_framework/exatest/__init__.py (hunk 1 above)
Comment (ckunki) [reply exasol_python_test_framework/exatest/__init__.py:250]: While the term "TLS" might be more correct than "SSL", I hesitate to introduce a renaming layer, here. 
The term used in file `odbc.ini` is coined to SSL. So, I fear, we might need to live with that.

**Review by kaklakariada — DISMISSED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**